src/App.tsx

```diff
@@ -1,61 +1,144 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+type User = {
+  id: number,
+  name: string,
+  username: string,
+  email: string,
+};
+
+type Todo = {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+  user: User,
+};
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || {
+    id: 0,
+    name: '',
+    username: '',
+    email: '',
+  };
+}
+
+const todos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId) || {
+    id: 0, name: '', username: '', email: '',
+  },
+}
+));
 
 export const App = () => {
+  // #region states
+
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>(todos);
+  const [currentUser, setCurrentUser] = useState(0);
+  const [selectedError, setSelectedUser] = useState(false);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  // #endregion
+  // #region functions
+
+  const addTodo = (newTodo: Todo) => {
+    if (newTodo) {
+      setCurrentTodos(curentTodos => [...curentTodos, newTodo]);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setHasTitleError(!todoTitle);
+    setSelectedUser(!currentUser);
+
+    if (currentUser === 0 || !todoTitle) {
```

Comment:
i can add the todo only with spaces, let's fix that
<img width="253" alt="image" src="https://github.com/mate-academy/react_add-todo-form/assets/70220667/84715d29-c8ef-4fb3-babb-410ed7fce2f3">

Reply:
Not fixed 
<img width="236" alt="Screenshot 2024-02-21 at 14 03 13" src="https://github.com/mate-academy/react_add-todo-form/assets/135232001/da3bd97e-7cdb-4fb4-825e-05f40dd7e999">